cmd/operator-sdk/run/local.go

```diff
@@ -76,63 +79,69 @@ func (c runLocalArgs) run() error {
 	return projutil.ErrUnknownOperatorType{}
 }
 
+// runGo will run the project locally for the Go Type projects
 func (c runLocalArgs) runGo() error {
-	projutil.MustInProjectRoot()
-	absProjectPath := projutil.MustGetwd()
-	projectName := filepath.Base(absProjectPath)
-	outputBinName := filepath.Join(scaffold.BuildBinDir, projectName+"-local")
-	if runtime.GOOS == "windows" {
-		outputBinName += ".exe"
-	}
-	if err := c.buildLocal(outputBinName); err != nil {
-		return fmt.Errorf("failed to build operator to run locally: %v", err)
-	}
-
-	args := []string{}
-	if c.operatorFlags != "" {
-		extraArgs := strings.Split(c.operatorFlags, " ")
-		args = append(args, extraArgs...)
+	// todo: remove the vars and if and else
```

Comment:
Seems to me like a bunch of extraneous changes in this file.

Can we just update `buildLocal` to take a path to the `main.go` file, and then chose the `main.go` file path based on `kbutil.HasProjectFile()`?

Reply:
Since I need to check how it works, I extracted the logic in funcs and then, added comments to make easier we understand and keep it maintained.

Reply:
It seems like there is a bunch of duplication though. Everything should be exactly the same for SDK go projects and KB go projects except for the path to the main.go file.

Reply:
Duplication? Did you check it whiteout the diff?

Reply:
All done. really tks for your input.